### PR TITLE
Persistence verification

### PR DESCRIPTION
**Pull request by sifraser**
With working XO example

Creating so you can have a provisional review on Monday

### CODE REVIEW

File: azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/ImplementationInstance.kt [hunk 1]
```diff
@@ -67,17 +96,12 @@ interface ImplementationInstance<
                     createInstancesFromConfiguredJars()
                 } else {
                     Log.trace("Loading implementation instances from classpath")
+                    if (havePersistenceVerificationInstance) {
+                        Log.warn("It not advised to mix classpath loading with the user of persistence verification instance. There may be unexpected classloading issues!")
```
Comment (donbex) [inline azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/ImplementationInstance.kt]: ⛏️ Looks like there might be a typo here.
```suggestion
                        Log.warn("It not advised to mix classpath loading with the use of persistence verification instance. There may be unexpected classloading issues!")
```

File: azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/TaskType.kt [hunk 2]
```diff
@@ -7,4 +7,5 @@ enum class TaskType {
     GenerateActions,
     Query,
     Verify,
+    PersistenceVerify,
```
Comment (donbex) [inline azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/TaskType.kt:10]: ⛏️ Why not `VerifyPersistence`?

File: tictactoe/implementation/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/implementation/GameManager.kt [hunk 3]
```diff
@@ -36,16 +39,28 @@ class GameManager(private val store: File) {
...
         Log.info("Active games = ${games.keys}")
         return game
     }
 
     private val objectMapper = ObjectMapper()
         .registerModule(KotlinModule())
 
+    val gameCreator: GameCreator by lazy {
+        val loader = ServiceLoader.load(GameCreator::class.java)
+        loader.iterator().asSequence().first()
```
Comment (donbex) [inline tictactoe/implementation/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/implementation/GameManager.kt]: ⛏️ Given that `first` also throws an exception when the sequence is empty, what's the benefit of this over
`loader.iterator().next()`?

**Review by donbex — COMMENTED** [review]
LGTM. Not approving yet since it is still a draft.

File: azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/TaskType.kt (hunk 2 above)
Comment (sifraser) [reply azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/TaskType.kt:10]: We have used the term 'persistence verification' throughout

File: tictactoe/implementation/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/implementation/GameManager.kt (hunk 3 above)
Comment (sifraser) [reply tictactoe/implementation/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/implementation/GameManager.kt]: It had simply been copied from another load where we  expected >=1

**Comment by sifraser** [comment]
This has now been used in anger at scale without finding any issue in the framework code, so we should now move to integrate this into develop whenever you're ready.

**Review by donbex — APPROVED** [review]

**Pull request merged**